website/blog/2019-12-09-beta-release.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Announcing Beta of component-test-utils
+author: Florent Berthelot
+authorURL: http://twitter.com/berthel350
+---
+
+I've been working on this side-project for almost 1 year now and I think it's mature enough to pass another step: The Beta step !
+
+I've never communicated about this library except to my closest co-workers and friends. So let's starts this year by releasing it !
+
+## Why another testing library for react ?
+
+There are two actors for testing react components:
+
+- [Enzyme](https://airbnb.io/enzyme/)
+- [React-testing-library](https://github.com/testing-library/react-testing-library)
+
+They are great and I've used them on various projects, but I think there is a gap between which is not filled.
+
+Enzyme has an API that forces you to test components one by one, which is really great when you think about making unit tests small, independent, performant. However, when you handle multiple components, the enzyme API is not well designed for that. You will have to add multiple `.dive()`, the rendered HTML tree is not well represented except when calling `.debug()` function.
+
+React-testing-library is the opposite of Enzyme. On one hand, it forces you not to mock anything. This way, you will test some patterns like HoC very easily, but on the other hand it's not very performant. Imagine if you want to test a component that has many many many sub-components, you will have two solutions: to use a mocking system provided by a tierce library (jest ?) or to not care about the time your unit test will take.
+
+With component-test-utils-react, you have a fine control where your mock starts and where it finishes. This way you won't have to handle many `.dive` in your app nor to deal with many mock `jest.mock('./anotherCmp', /**/)`.
+
+Here is an example of testing with component-test-utils:
+
+```jsx
+const LikeButton = props => {
+  return <button {...props}>+</button>;
+};
+
+const LikeDisplayer = () => {
+  const [nbLikes, setNbLikes] = React.useState(0);
+  const clickHandler = () => setNbLikes(nbLikes + 1);
+  return (
+    <p>
+      {nbLikes} Like <LikeButton onClick={clickHandler} />
+    </p>
+  );
+};
+
+const component = shallow(<LikeDisplayer />, {
+  mocks: {
+    LikeButton
+  }
+});
+
+component.querySelector("button").dispatchEvent("click");
+
+expect(component.html()).toBe(
+  '<p>1 Like <LikeButton onClick="[clickHandler]"><button onClick="[clickHandler]">+</button></LikeButton></p>'
+);
+```
+
+## Only react... for now !
+
+When I started Component-test-utils I designed the API for testing components whichever the Framework used.
+
+But then why support only React ? Because, I had a very busy year 2019 with a lot of stuff to do like starting up my own company: WeFacto (there is no website for now). I've focused on React for now, in order to stabilize the API.
```

Comment:
```suggestion
But then why support only React ? Because, I had a very busy year 2019 with a lot of stuff to do like starting up my own company: WeFacto (there is no website for now). I've focused on React for the time being, in order to stabilize the API.
```

(Repetition)